### PR TITLE
Get apis of Applicants and Jobs

### PR DESCRIPTION
**Pull request by gem-charugarg**

### CODE REVIEW

**Review by gem-aryanpurru — COMMENTED** [review]
1. Add logger info in catch block
2. Add proper comments before function definition
3. Follow correct camel casing while naming function. 
4. If condition for ((response.getResponseBody()) != null) must be added to line 75 (utils file) to avoid, redundant code.
5. Feature file contains multiple redundant code
6. Remove all the redundant comments with code.

**Review by gem-pawandeep — CHANGES_REQUESTED** [review]
1. change stepdefinition to stepDefinition.
2. remove unwanted variables.
3. remove dead code.
4. use ^ $ in   @Given("Set the Applicant endpoint {string} and method {string}").
5. use appropriate language for scenario name, ex : ATS, Healthcheck for Applicants API.
6. use proper camel casing in method an variables name

**Comment by gem-charugarg** [comment]
> 1. change stepdefinition to stepDefinition.
> 2. remove unwanted variables.
> 3. remove dead code.
> 4. use ^ $ in   @given("Set the Applicant endpoint {string} and method {string}").
> 5. use appropriate language for scenario name, ex : ATS, Healthcheck for Applicants API.
> 6. use proper camel casing in method an variables name
 

HI @gem-pawandeep, I will make changes accordingly. but there are few points to consider from my end as well.
As per the 1st point, I named package as per accepted industry standard (Attaching the reference link below)
[Oracle package naming conventions](https://docs.oracle.com/javase/tutorial/java/package/namingpkgs.html)
Also for 2nd ,3rd and 6th points you might have not reviewed the latest commit, it might be the previous commit.
And as per the remaining points I will make changes and will update you soon.
Thanks.

File: src/main/java/com/qa/ats/stepdefinition/ApplicantStep.java [hunk 1]
```diff
@@ -1,28 +1,42 @@
...
     @Given("Set the Applicant endpoint {string} and method {string}")
     public void setThePolicyEndpointAndMethod(String url, String method) {
         HashMap<String, String> token = new HashMap<String, String>();
-        //token.put("X-REMOTE-USER-EMAIL", "saru.goyal@geminisolutions.com");
         try {
-            status = Utils.APIwithoutPayloads(url, method, token, "").getStatus();
-        } catch (Exception e) {
-            throw new RuntimeException(e);
+            status = Utils.apiWithoutPayloads(url, method, token, "").getStatus();
+            GemTestReporter.addTestStep("Hit API", "User not able to hit the API", STATUS.PASS);
```
Comment (gem-pawandeep) [inline src/main/java/com/qa/ats/stepdefinition/ApplicantStep.java:23]: I think this line shouldn't be there. GemTestReporter.addTestStep("Hit API", "User not able to hit the API", STATUS.PASS)

File: src/main/java/com/qa/ats/stepdefinition/JobsStep.java [hunk 2]
```diff
@@ -1,30 +1,44 @@
...
         }
     }
 
+    //Verify the status
     @Then("Verify Job status code {int}")
     public void verifyPolicyStatusCodeExpectedStatus(Integer Expected) {
-        Utils.VerifyStatusCode(Expected, status);
+        try {
+            Utils.verifyStatusCode(Expected, status);
+            GemTestReporter.addTestStep("Status Check", "Verify the API status", STATUS.PASS);
```
Comment (gem-pawandeep) [inline src/main/java/com/qa/ats/stepdefinition/JobsStep.java:37]: if you are verifying the status code through verifyStatusCode method then GemTestReporter.addTestStep("Status Check", "Verify the API status", STATUS.PASS); shouldn't be there

File: src/main/java/com/qa/ats/utils/Utils.java [hunk 3]
```diff
@@ -30,31 +38,27 @@ public static Response APIwithoutPayloads(String UrlNameFromConfig, String metho
...
             GemTestReporter.addTestStep("Response Message", response.getResponseMessage(), STATUS.INFO);
-            if ((response.getResponseBody()) != null) {
-                GemTestReporter.addTestStep("Response Body", response.getResponseBody(), STATUS.INFO);
-            } else {
-                GemTestReporter.addTestStep("Response Body", "No-Response", STATUS.INFO);
-            }
+            responseCheck(response);
         } catch (Exception e) {
             GemTestReporter.addTestStep(method.toUpperCase() + " Request Verification ", method.toUpperCase() + " Request Did not Executed Successfully", STATUS.FAIL);
             GemTestReporter.addTestStep("Response Message", response.getResponseMessage(), STATUS.INFO);
```
Comment (gem-pawandeep) [inline src/main/java/com/qa/ats/utils/Utils.java:46]: please combine or change these 2 steps and use logger here,

**Review by gem-pawandeep — APPROVED** [review]

**Review by gem-pawandeep — APPROVED** [review]
approved!!!

**Pull request merged**